Description: Enable all compilation flags, including related to hardening
Author: Lucas Kanashiro <lucas.kanashiro@collabora.co.uk>
Last-Updated: 2018-08-31

--- a/Makefile
+++ b/Makefile
@@ -25,6 +25,7 @@ PAMFLAGS = $(LDFLAGS) $(INCLUDE_FLAGS) -
 NSSFLAGS = $(LDFLAGS) $(INCLUDE_FLAGS) -shared -Wl,-soname,$(NSS_LIBRARY_SONAME)
 LIBNSSFLAGS = $(LDFLAGS) -Wall -Wstrict-prototypes -fPIC
 LIBNSS_SO_FLAGS = $(LIBNSSFLAGS) -shared -Wl,-soname,$(LIBNSS_CACHE_OSLOGIN_SONAME)
+ALLFLAGS = $(CFLAGS) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS)
 
 # UTILS
 UTILS_DIR = utils
@@ -92,40 +93,40 @@ all: $(NSS) $(NSS_CACHE_BIN) $(LIBNSS_CA
 endif
 
 $(NSS): $(NSS_LIBRARY_SOURCE) $(UTILS)
-	$(CXX) $(CXXFLAGS) $(NSSFLAGS) -o $(NSS_LIBRARY_NAME) \
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) $(NSSFLAGS) -o $(NSS_LIBRARY_NAME) \
 	$(NSS_SRC) $(UTILS) $(LIBS)
 
 $(NSS_CACHE_BIN): $(NSS_CACHE_SRC) $(UTILS_SRC)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(INCLUDE_FLAGS) -o $(NSS_CACHE_BIN) $(NSS_CACHE_SRC) $(UTILS_SRC) $(LIBS)
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) $(LDFLAGS) $(INCLUDE_FLAGS) -o $(NSS_CACHE_BIN) $(NSS_CACHE_SRC) $(UTILS_SRC) $(LIBS)
 
 $(LIBNSS_CACHE_OSLOGIN_NAME): $(LIBNSS_CACHE_OBJ) $(LIBNSS_COMPAT_OBJ)
-	$(CXX) $(LIBNSS_SO_FLAGS) -o $(LIBNSS_CACHE_OSLOGIN_NAME) $(LIBNSS_CACHE_OBJ) $(LIBNSS_COMPAT_OBJ)
+	$(CXX) $(ALLFLAGS) $(LIBNSS_SO_FLAGS) -o $(LIBNSS_CACHE_OSLOGIN_NAME) $(LIBNSS_CACHE_OBJ) $(LIBNSS_COMPAT_OBJ)
 
 $(LIBNSS_CACHE_OBJ): $(LIBNSS_CACHE_SRC)
-	$(CC) $(LIBNSSFLAGS) -c -o $(LIBNSS_CACHE_OBJ) $(LIBNSS_CACHE_SRC)
+	$(CC) $(ALLFLAGS) $(LIBNSSFLAGS) -c -o $(LIBNSS_CACHE_OBJ) $(LIBNSS_CACHE_SRC)
 
 $(LIBNSS_COMPAT_OBJ): $(LIBNSS_COMPAT_SRC)
-	$(CC) $(LIBNSSFLAGS) -c -o $(LIBNSS_COMPAT_OBJ) $(LIBNSS_COMPAT_SRC)
+	$(CC) $(ALLFLAGS) $(LIBNSSFLAGS) -c -o $(LIBNSS_COMPAT_OBJ) $(LIBNSS_COMPAT_SRC)
 
 $(PAM): $(PAM_ADMIN_MOD) $(PAM_LOGIN_MOD)
 
 $(PAM_LOGIN_MOD): $(PAM_LOGIN_OBJ) $(UTILS)
-	$(CXX) $(PAMFLAGS) -o $(PAM_LOGIN_MOD) $(PAM_LOGIN_OBJ) $(UTILS) $(PAM_LIBS)
+	$(CXX) $(ALLFLAGS) $(PAMFLAGS) -o $(PAM_LOGIN_MOD) $(PAM_LOGIN_OBJ) $(UTILS) $(PAM_LIBS)
 
 $(PAM_ADMIN_MOD): $(PAM_ADMIN_OBJ) $(UTILS)
-	$(CXX) $(PAMFLAGS) -o $(PAM_ADMIN_MOD) $(PAM_ADMIN_OBJ) $(UTILS) $(PAM_LIBS)
+	$(CXX) $(ALLFLAGS) $(PAMFLAGS) -o $(PAM_ADMIN_MOD) $(PAM_ADMIN_OBJ) $(UTILS) $(PAM_LIBS)
 
 $(PAM_LOGIN_OBJ): $(PAM_LOGIN_SRC)
-	$(CXX) $(CXXFLAGS) -c $(PAM_LOGIN_SRC) -o $(PAM_LOGIN_OBJ)
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) -c $(PAM_LOGIN_SRC) -o $(PAM_LOGIN_OBJ)
 
 $(PAM_ADMIN_OBJ): $(PAM_ADMIN_SRC)
-	$(CXX) $(CXXFLAGS) -c $(PAM_ADMIN_SRC) -o $(PAM_ADMIN_OBJ)
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) -c $(PAM_ADMIN_SRC) -o $(PAM_ADMIN_OBJ)
 
 $(AUTHKEYS_BIN): $(AUTHKEYS_SRC) $(UTILS_SRC)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(INCLUDE_FLAGS) -o $(AUTHKEYS_BIN) $(AUTHKEYS_SRC) $(UTILS_SRC) $(LIBS)
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) $(LDFLAGS) $(INCLUDE_FLAGS) -o $(AUTHKEYS_BIN) $(AUTHKEYS_SRC) $(UTILS_SRC) $(LIBS)
 
 $(UTILS): $(UTILS_SRC)
-	$(CXX) $(CXXFLAGS) $(INCLUDE_FLAGS) -c $(UTILS_SRC) -o $(UTILS)
+	$(CXX) $(ALLFLAGS) $(CXXFLAGS) $(INCLUDE_FLAGS) -c $(UTILS_SRC) -o $(UTILS)
 
 $(SELINUX_MOD_FILE): $(SELINUX_MODULE_SRC)
 	checkmodule -M -m -o $(SELINUX_MOD_FILE) $(SELINUX_MODULE_SRC)
